File: server/src/modules/user/model/user.model.js
const userModel = {
    table: "users",
    field: {
        userId: "user_id",
        fullName: "full_name",
        email: "email",
        phone: "phone",
        role: "role",

        isVerified: "is_verified", // Xác thực tài khoản

        status: "status",
        banReason: "ban_reason",
        bannedAt: "banned_at",
        bannedBy: "banned_by",
        avatarUrl: "avatar_url",
        createdAt: "created_at",
        updatedAt: "updated_at"
    },
};

module.exports = userModel;

File: mobile/lib/features/auth/models/user_model.dart
import 'package:mobile/features/auth/models/rescuer_model.dart';

class UserModel {
  final String userId;
  final String? email;
  final String? phone;
  final String role;
  final bool isVerified;
  final String status;
  final String? banReason;
  final String? bannedAt;
  final RescuerModel? rescuer;

  UserModel({
    required this.userId,
    this.email,
    this.phone,
    required this.role,
    required this.isVerified,
    required this.status,
    this.banReason,
    this.bannedAt,
    this.rescuer,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      userId: json['userId'] ?? json['user_id'] ?? '',
      email: json['email'],
      phone: json['phone'],
      role: json['role'] ?? '',
      isVerified: json['isVerified'] ?? json['is_verified'] ?? false,
      status: json['status'] ?? '',
      banReason: json['banReason'] ?? json['ban_reason'],
      bannedAt: json['bannedAt'] ?? json['banned_at'],
      rescuer: json.containsKey('rescuer') && json['rescuer'] != null
          ? RescuerModel.fromJson(json['rescuer'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'role': role,
      // ... bỏ các thuộc tính khác của bạn vào đây tương tự
    };
  }
}